frontend/ui/widget/timewidget.lua

```diff
@@ -0,0 +1,354 @@
+    }
+
+    local text_hours = TextBoxWidget:new{
+        text = string.format("%02d", self.curr_hour),
+        alignment = "center",
+        face = self.title_face,
+        text_font_size = 24,
+        bold = true,
+        width = self.screen_width * 0.20,
+    }
+    local text_minutes = TextBoxWidget:new{
+        text = string.format("%02d", self.curr_min),
+        alignment = "center",
+        face = self.title_face,
+        text_font_size = 24,
+        bold = true,
+        width = self.screen_width * 0.20,
+    }
+
+    local colon_space = TextBoxWidget:new{
+        text = ":",
+        alignment = "center",
+        face = self.title_face,
+        bold = true,
+        width = self.screen_width * 0.20 + 2 * button_up_hours.bordersize + 2 * button_up_minutes.bordersize
+    }
+
+    local button_table_up = HorizontalGroup:new{
+        align = "center",
+        button_up_hours,
+        empty_space,
+        button_up_minutes,
+    }
+    local time_text_table = HorizontalGroup:new{
+        align = "center",
+        text_hours,
+        colon_space,
+        text_minutes,
+    }
+    local button_table_down = HorizontalGroup:new{
+        align = "center",
+        button_down_hours,
+        empty_space,
+        button_down_minutes,
+    }
+
+    table.insert(button_group_up, button_table_up)
+    table.insert(button_group_down, button_table_down)
+    table.insert(vertical_group, padding_span_top_bottom)
+    table.insert(vertical_group, button_group_up)
+    table.insert(vertical_group, padding_span)
+    table.insert(vertical_group, time_text_table)
+    table.insert(vertical_group, padding_span)
+    table.insert(vertical_group, button_group_down)
+    table.insert(vertical_group, padding_span_top_bottom)
+
+    return CenterContainer:new{
+        dimen = Geom:new{
+            w = self.screen_width * 0.95,
+            h = vertical_group:getSize().h
+        },
+        vertical_group
+    }
+end
+
+function TimeWidget:update()
+    local time_title = FrameContainer:new{
+        padding = Screen:scaleBySize(5),
+        margin = Screen:scaleBySize(2),
+        bordersize = 0,
+        TextWidget:new{
+            text = self.title_text,
+            face = self.title_face,
+            bold = true,
+            width = self.screen_width * 0.95,
+        },
+    }
+    local time_container = FrameContainer:new{
+        padding = Screen:scaleBySize(2),
+        margin = Screen:scaleBySize(2),
+        bordersize = 0,
+        self:paintContainer()
+    }
+    local time_line = LineWidget:new{
+        dimen = Geom:new{
+            w = self.width,
+            h = Screen:scaleBySize(2),
+        }
+    }
+    local time_bar = OverlapGroup:new{
+        dimen = {
+            w = self.width,
+            h = time_title:getSize().h
+        },
+        time_title,
+        CloseButton:new{ window = self, },
+    }
+    local buttons = {
+        {
+            {
+                text = "OK",
```

Comment:
The UX style is Cancel left, OK right. Also keep in mind the UX guidelines: http://koreader.rocks/doc/modules/ui.widget.confirmbox.html

> It is strongly recommended to set a custom `ok_text` describing the action to be confirmed.

Seeing how the widget should be generic OK is fine here as a default, but the widget should preferably implement an `ok_text` and such similar to other widgets.

https://github.com/koreader/koreader/blob/30378eb2a82ac1b2f686b0db638b194b642af4ab/frontend/ui/widget/confirmbox.lua#L43-L46

Then where you call it (in common_menu) add an `ok_text = "Set time"` :+1: